jetcd-core/src/test/java/io/etcd/jetcd/WatchTest.java

```diff
@@ -16,244 +16,193 @@
 
 import io.etcd.jetcd.Watch.Watcher;
 import io.etcd.jetcd.common.exception.CompactedException;
 import io.etcd.jetcd.kv.PutResponse;
-import io.etcd.jetcd.launcher.junit.EtcdClusterResource;
+import io.etcd.jetcd.launcher.junit5.EtcdClusterExtension;
 import io.etcd.jetcd.options.WatchOption;
 import io.etcd.jetcd.watch.WatchEvent;
 import io.etcd.jetcd.watch.WatchEvent.EventType;
 import io.etcd.jetcd.watch.WatchResponse;
+
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
 
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.Timeout;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 /**
  * watch test case.
  */
-@RunWith(Parameterized.class)
 public class WatchTest {
-  @ClassRule
-  public static final EtcdClusterResource clusterResource = new EtcdClusterResource("watch", 3);
+  @RegisterExtension
+  public static final EtcdClusterExtension cluster = new EtcdClusterExtension("watch", 3);
   private static final ByteSequence namespace = bytesOf("test-namespace/");
 
-  private static Client client;
-  private static Client clientWithNamespace;
-  private static KV kvClient;
-
-  private boolean useNamespace;
-  private Watch watchClient;
-
-  @Rule
-  public Timeout timeout = Timeout.seconds(10);
+  private Duration timeout = Duration.ofSeconds(10);
 
-  public WatchTest(boolean useNamespace) {
-    this.useNamespace = useNamespace;
-    this.watchClient = useNamespace ? clientWithNamespace.getWatchClient()
-        : client.getWatchClient();
+  static Stream<Arguments> parameters() {
+    return Stream.of(
+        arguments(Client.builder().endpoints(cluster.getClientEndpoints()).namespace(namespace).build(),
+            ByteSequence.from(namespace.getByteString().concat(randomByteSequence().getByteString()))),
+        arguments(Client.builder().endpoints(cluster.getClientEndpoints()).build(),
+            randomByteSequence())
+    );
   }
 
-  @Parameterized.Parameters(name = "UseNamespace = {0}")
-  public static List<Boolean> parameters() {
-    return Arrays.asList(true, false);
-  }
-
-  @BeforeClass
-  public static void setUp() {
-    client = Client.builder().endpoints(clusterResource.cluster().getClientEndpoints()).build();
-    clientWithNamespace = Client.builder().endpoints(clusterResource.cluster()
-        .getClientEndpoints()).namespace(namespace).build();
-    kvClient = client.getKVClient();
```

Comment:
@fanminshi was that intentional to take `kvClient` from `client` (not namespaced)?

Reply:
@yb172 before namespace added. it was there just a `kvClient` that you can use to put `KV` into etcd. So instead of calling `client.GetKVClient()` all the time, you can just call `kvClient` instead. After namespace pr, it was used in `putKV()` helper function to add key with namespace or not into KV. After a second thought, it was kinda of hacky. However, I think you refactor achieve the same result more elegantly.